internal/quantile/aggregate.go

```diff
@@ -74,36 +74,43 @@ func (e *E2eProcessingLatencyAggregate) Less(i, j int) bool {
 	return e.Percentiles[i]["percentile"] > e.Percentiles[j]["percentile"]
 }
 
+// Add merges two E2eProcessingLatencyAggregate by averaging the percentiles
+//
+// It is side-effect free and does not modify the receiver or the parameter
```

Comment:
I would be fine with this approach, too (as it would also now be internally consistent).

... and I thought this would be possible too, even if `x` is initially `nil`, and I might be missing something simple and obvious, but I couldn't seem to get it to work the way I would have expected.